lib/services/database_test.go

```diff
@@ -485,3 +487,249 @@ func TestDatabaseFromRedshiftCluster(t *testing.T) {
+			AccountID: "1234567890",
+			Region:    "us-east-1",
+			ElastiCache: types.ElastiCache{
+				ReplicationGroupID:       "my-cluster",
+				UserGroupIDs:             []string{"my-user-group"},
+				TransitEncryptionEnabled: true,
+				EndpointType:             awsutils.ElastiCacheReaderEndpoint,
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	actual, err := NewDatabasesFromElastiCacheNodeGroups(cluster, extraLabels)
+	require.NoError(t, err)
+	require.Equal(t, types.Databases{expectedPrimary, expectedReader}, actual)
+}
+
+func TestExtraElastiCacheLabels(t *testing.T) {
+	cluster := &elasticache.ReplicationGroup{
+		ReplicationGroupId: aws.String("my-redis"),
+	}
+	tags := []*elasticache.Tag{
+		{Key: aws.String("key1"), Value: aws.String("value1")},
+		{Key: aws.String("key2"), Value: aws.String("value2")},
+	}
+
+	nodes := []*elasticache.CacheCluster{
+		{
+			ReplicationGroupId:   aws.String("some-other-redis"),
+			EngineVersion:        aws.String("8.8.8"),
+			CacheSubnetGroupName: aws.String("some-other-subnet-group"),
+		},
+		{
+			ReplicationGroupId:   aws.String("my-redis"),
+			EngineVersion:        aws.String("6.6.6"),
+			CacheSubnetGroupName: aws.String("my-subnet-group"),
+		},
+	}
+
+	subnetGroups := []*elasticache.CacheSubnetGroup{
+		{
+			CacheSubnetGroupName: aws.String("some-other-subnet-group"),
+			VpcId:                aws.String("some-other-vpc"),
+		},
+		{
+			CacheSubnetGroupName: aws.String("my-subnet-group"),
+			VpcId:                aws.String("my-vpc"),
+		},
+	}
+
+	tests := []struct {
+		name              string
+		inputTags         []*elasticache.Tag
+		inputNodes        []*elasticache.CacheCluster
+		inputSubnetGroups []*elasticache.CacheSubnetGroup
+		expectLabels      map[string]string
+	}{
+		{
+			name:              "all tags",
+			inputTags:         tags,
+			inputNodes:        nodes,
+			inputSubnetGroups: subnetGroups,
+			expectLabels: map[string]string{
+				"key1":           "value1",
+				"key2":           "value2",
+				"engine-version": "6.6.6",
+				"vpc-id":         "my-vpc",
+			},
+		},
+		{
+			name:              "no resource tags",
+			inputTags:         nil,
+			inputNodes:        nodes,
+			inputSubnetGroups: subnetGroups,
+			expectLabels: map[string]string{
+				"engine-version": "6.6.6",
+				"vpc-id":         "my-vpc",
+			},
+		},
+		{
+			name:              "no nodes",
+			inputTags:         tags,
+			inputNodes:        nil,
+			inputSubnetGroups: subnetGroups,
+
+			// Without subnet group name from nodes, VPC ID cannot be found.
+			expectLabels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+		},
+		{
+			name:              "no subnet groups",
+			inputTags:         tags,
+			inputNodes:        nodes,
+			inputSubnetGroups: nil,
+			expectLabels: map[string]string{
+				"key1":           "value1",
+				"key2":           "value2",
+				"engine-version": "6.6.6",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
`t.Parallel()` ?

Reply:
I've thought about and I actually had `t.Parallel` initially then later removed it. There is no chan or goroutine here. And it's very quick function. So parallel may not provide much benefit.